packages/web3-utils/src/converters.ts

```diff
@@ -411,6 +412,7 @@ export const toHex = (
  */
 export const toNumber = (value: Numbers): number | bigint => {
 	if (typeof value === 'number') {
+			console.warn('Warning: Using type `number` with values that are large or contain many decimals may cause loss of precision, it is recommended to use type `string` or `BigInt` when using conversion methods')
```

Comment:
I added a warning to `toNumber`, I think it would be good overall that users should be using bigint or strings to convert but it may be alot of warnings that a user will face if they use this method multiple times. thoughts?

Reply:
I think it can be an annoying warning. What do you think if we add this warning they use a number larger than max?

Reply:
good suggestion thanks!